### PR TITLE
Add filter by subject

### PR DESCRIPTION
**Pull request by simonwhatley**
Fixes the following issue: https://trello.com/c/YIWvakTK/3379-applications-list-filter-by-subject

### CODE REVIEW

File: app/routes/application-list.js [hunk 1]
```diff
@@ -332,6 +361,18 @@ module.exports = router => {
         })
       }
 
+      if (subjects && subjects.length) {
+        selectedFilters.categories.push({
+          heading: { text: 'Subjects' },
```
Comment (adamsilver) [inline app/routes/application-list.js:366]: `Subject`

File: app/routes/application-list.js [hunk 2]
```diff
@@ -375,7 +418,8 @@ module.exports = router => {
       applications,
       pagination,
       selectedFilters,
-      hasFilters
+      hasFilters,
+      subjectItems
```
Comment (adamsilver) [inline app/routes/application-list.js:422]: Why do we need this?

Comment (simonwhatley) [reply app/routes/application-list.js]: We derive subjects from data and pass them to the view to populate the subject filter.

File: app/routes/application-list.js (hunk 1 above)
Comment (simonwhatley) [reply app/routes/application-list.js:366]: Can you clarify whether you want the plural or singular form of 'subject'? I assume the singular, but other filter headings are plural like `Providers` and `Statuses`.

**Pull request merged**